### PR TITLE
Convert scan azimuth to turret coords

### PR DESCRIPTION
**Pull request by raulvera59**
Previously slew_to_azimuth did the conversion, but the scan azimuth was still wrong. This refactors out a conversion function and has both places call it.

### CODE REVIEW

File: components/turret.py [hunk 1]
```diff
@@ -126,24 +126,31 @@ def setup(self) -> None:
...
 
+    # Convert an angle in the robot coordinate system to the turret coordinate
+    # system, which is rotated by 180 degrees (i.e. 0 is backwards on the robot).
+    def robot_to_turret(self, angle: float) -> float:
+        turret_angle = angle - math.pi  # This converts to turret coordinate system
+        # Now we normalise to +- pi
+        while turret_angle < -math.pi:
+            turret_angle += math.tau
+        while turret_angle > math.pi:
+            turret_angle -= math.tau
```
Comment (auscompgeek) [inline components/turret.py]: Wouldn't it be simpler to use some trig to do this? (i.e. take the arc tangent of a sine and cosine of the angle)

... we could even use Rotation2d operations to do this

File: components/turret.py [hunk 2]
```diff
@@ -126,24 +126,31 @@ def setup(self) -> None:
...
+            turret_angle = self.robot_to_turret(angle)
+            self.motor._slew_to_counts(
+                int(turret_angle * self.COUNTS_PER_TURRET_RADIAN)
+            )
         else:
             print("slew_to_azimuth() called before index found")
 
+    # Convert an angle in the robot coordinate system to the turret coordinate
+    # system, which is rotated by 180 degrees (i.e. 0 is backwards on the robot).
+    def robot_to_turret(self, angle: float) -> float:
```
Comment (auscompgeek) [inline components/turret.py]: Given that this function doesn't actually use self, should it be a method? (We could make it a staticmethod I guess.)

File: components/turret.py (hunk 1 above)
Comment (raulvera59) [reply components/turret.py]: No. It wouldn't be simpler.

File: components/turret.py (hunk 2 above)
Comment (raulvera59) [reply components/turret.py]: Done.

File: components/turret.py (hunk 1 above)
Comment (auscompgeek) [reply components/turret.py]: The while loops just strike me as odd I guess. We've had a `normalise_angle` that does `math.atan2(math.sin(theta), math.cos(theta))` in past years.

**Pull request merged**